### Data import and initial data cleaning

In [1]:
# import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# read data file and assign it to df
path = r"C:\Users\soonn\Documents\GitHub\CoffeeBeanSales\DatasetForCoffeeSales2.csv" # assign file path to 'path'
df = pd.read_csv(path) # load data and assign data frame to df


In [3]:
# initially check 5 rows of data to have an idea of data.
df.head()

Date  Customer_ID    City      Category     Product  Unit Price  \
0  1/1/2023           32  Riyadh  coffee beans   Colombian          40   
1  1/2/2023           49    Abha  coffee beans  Costa Rica          35   
2  1/3/2023           75   Tabuk  coffee beans  Costa Rica          35   
3  1/4/2023           80    Abha  coffee beans   Ethiopian          45   
4  1/5/2023           78    Hail  coffee beans   Colombian          40   

   Quantity  Sales Amount  Used_Discount  Discount_Amount  Final Sales  
0        14           560          False                0          560  
1        17           595          False                0          595  
2        19           665          False                0          665  
3         1            45          False                0           45  
4        46          1840           True              368         1472

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 730 entries, 0 to 729
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Date             730 non-null    object
 1   Customer_ID      730 non-null    int64 
 2   City             730 non-null    object
 3   Category         730 non-null    object
 4   Product          730 non-null    object
 5   Unit Price       730 non-null    int64 
 6   Quantity         730 non-null    int64 
 7   Sales Amount     730 non-null    int64 
 8   Used_Discount    730 non-null    bool  
 9   Discount_Amount  730 non-null    int64 
 10  Final Sales      730 non-null    int64 
dtypes: bool(1), int64(6), object(4)
memory usage: 57.9+ KB


In [5]:
df.shape

(730, 11)

In [6]:
df.size

8030

# Insight
- Data type of 'Date' is objects. Changing data type to datetime is better.
- There is no missing values in the data and it has shpe of 730 rows and 11 columns.
- there is 8030 values in the data set.


In [7]:
df.describe(include = 'all')

Date  Customer_ID  City      Category     Product  Unit Price  \
count          730   730.000000   730           730         730  730.000000   
unique         730          NaN    10             1           5         NaN   
top     12/30/2024          NaN  Hail  coffee beans  Costa Rica         NaN   
freq             1          NaN    87           730         165         NaN   
mean           NaN    51.669863   NaN           NaN         NaN   36.794521   
std            NaN    29.014339   NaN           NaN         NaN    4.955104   
min            NaN     1.000000   NaN           NaN         NaN   30.000000   
25%            NaN    27.250000   NaN           NaN         NaN   35.000000   
50%            NaN    52.000000   NaN           NaN         NaN   35.000000   
75%            NaN    77.000000   NaN           NaN         NaN   40.000000   
max            NaN   100.000000   NaN           NaN         NaN   45.000000   

          Quantity  Sales Amount Used_Discount  Discount_Amount  Final Sales  
count   730.000000    730.000000           730       730.000000   730.000000  
unique         NaN           NaN             2              NaN          NaN  
top            NaN           NaN          True              NaN          NaN  
freq           NaN           NaN           371              NaN          NaN  
mean     26.080822    959.924658           NaN        97.393151   862.531507  
std      14.480971    551.282730           NaN       123.401350   509.032315  
min       1.000000     30.000000           NaN         0.000000    24.000000  
25%      14.000000    495.000000           NaN         0.000000   448.000000  
50%      27.000000    960.000000           NaN        13.000000   840.000000  
75%      39.000000   1400.000000           NaN       192.000000  1260.000000  
max      49.000000   2205.000000           NaN       441.000000  2205.000000

# Insight
- 'City' has 10 unique values, which means 10 cities are sampled and the city 'Hail' counted 87 showing in 730 cities.
- 'Category' contain only one unique value.
- 'Unit Price' range is from 30 to 45. the spread is narrow.
- 'Quantity' range is from 1 to 49. this spread is wide compare to unit price.
- Almost half of sales have discounts on sales.
- Sold in discounted price, so final sales price is settled.

### Goals of EDA
1. understand the data
2. find possible sales increase opportunities
3. build predictive model for 2025 year to optimize inventory management.
4. 

In [8]:
# import the second data set
path = r"C:\Users\soonn\Documents\GitHub\CoffeeBeanSales\saudi_cities_geocoding.csv"
df2 = pd.read_csv(path)


In [9]:
# take a look of data structure
df2.head(10)

City  Latitude  Longitude
0    Riyadh   24.7136    46.6753
1    Jeddah   21.2854    39.2376
2     Mecca   21.3891    39.8579
3    Medina   24.5247    39.5692
4    Dammam   26.3927    49.9777
5    Khobar   26.2172    50.1971
6      Abha   18.2465    42.5117
7     Tabuk   28.3835    36.5662
8      Hail   27.5114    41.7208
9  Buraidah   26.3259    43.9740

### Insight
the dataset contains geographical information of 10 cities in Saudi Arabia. will combine df and df2 into one data set.


In [10]:
# create a new data 'df_combined' by merging df and df2 on the same city name.
df_combined = df.merge(df2, on = 'City')

In [11]:
df_combined.head(10)

Date  Customer_ID      City      Category     Product  Unit Price  \
0   1/1/2023           32    Riyadh  coffee beans   Colombian          40   
1   1/2/2023           49      Abha  coffee beans  Costa Rica          35   
2   1/3/2023           75     Tabuk  coffee beans  Costa Rica          35   
3   1/4/2023           80      Abha  coffee beans   Ethiopian          45   
4   1/5/2023           78      Hail  coffee beans   Colombian          40   
5   1/6/2023            1    Khobar  coffee beans   Colombian          40   
6   1/7/2023           95  Buraidah  coffee beans   Brazilian          30   
7   1/8/2023           27    Medina  coffee beans   Colombian          40   
8   1/9/2023           73      Hail  coffee beans   Guatemala          35   
9  1/10/2023           82    Dammam  coffee beans  Costa Rica          35   

   Quantity  Sales Amount  Used_Discount  Discount_Amount  Final Sales  \
0        14           560          False                0          560   
1        17           595          False                0          595   
2        19           665          False                0          665   
3         1            45          False                0           45   
4        46          1840           True              368         1472   
5        32          1280           True              256         1024   
6         1            30           True                6           24   
7        33          1320          False                0         1320   
8        47          1645          False                0         1645   
9         4           140          False                0          140   

   Latitude  Longitude  
0   24.7136    46.6753  
1   18.2465    42.5117  
2   28.3835    36.5662  
3   18.2465    42.5117  
4   27.5114    41.7208  
5   26.2172    50.1971  
6   26.3259    43.9740  
7   24.5247    39.5692  
8   27.5114    41.7208  
9   26.3927    49.9777

In [12]:
df_combined.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 730 entries, 0 to 729
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Date             730 non-null    object 
 1   Customer_ID      730 non-null    int64  
 2   City             730 non-null    object 
 3   Category         730 non-null    object 
 4   Product          730 non-null    object 
 5   Unit Price       730 non-null    int64  
 6   Quantity         730 non-null    int64  
 7   Sales Amount     730 non-null    int64  
 8   Used_Discount    730 non-null    bool   
 9   Discount_Amount  730 non-null    int64  
 10  Final Sales      730 non-null    int64  
 11  Latitude         730 non-null    float64
 12  Longitude        730 non-null    float64
dtypes: bool(1), float64(2), int64(6), object(4)
memory usage: 69.3+ KB


In [13]:
# check data duplication.
df_duplicated = df_combined[df_combined.duplicated()]
df_duplicated

Empty DataFrame
Columns: [Date, Customer_ID, City, Category, Product, Unit Price, Quantity, Sales Amount, Used_Discount, Discount_Amount, Final Sales, Latitude, Longitude]
Index: []

### Insight
- The combined dataset does not have duplication. 

In [14]:
# change data type of 'date'
df_combined['Date'] = pd.to_datetime(df_combined['Date'])
df_combined.info() # check data type changed successfully.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 730 entries, 0 to 729
Data columns (total 13 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   Date             730 non-null    datetime64[ns]
 1   Customer_ID      730 non-null    int64         
 2   City             730 non-null    object        
 3   Category         730 non-null    object        
 4   Product          730 non-null    object        
 5   Unit Price       730 non-null    int64         
 6   Quantity         730 non-null    int64         
 7   Sales Amount     730 non-null    int64         
 8   Used_Discount    730 non-null    bool          
 9   Discount_Amount  730 non-null    int64         
 10  Final Sales      730 non-null    int64         
 11  Latitude         730 non-null    float64       
 12  Longitude        730 non-null    float64       
dtypes: bool(1), datetime64[ns](1), float64(2), int64(6), object(3)
memory usage: 69.3+ KB


### Upcoming
 - Duplication check is done
 - Data type changed
 - No visible missing values in dat